java/catalog-adapter-hms/src/main/java/com/lancedb/catalog/adapter/hms/service/HmsCatalogService.java

```diff
@@ -0,0 +1,333 @@
+        if (hmsTbl != null) {
+          String dsLocation = Utils.renameScheme(hmsTbl.getSd().getLocation());
+          LOGGER.info("Creating dataset: {}.{} under {}", dbName, tableName, location);
+
+          return Dataset.create(
+              allocator,
+              dsLocation,
+              schema,
+              new WriteParams.Builder().withStorageOptions(this.storageOption).build());
+        } else {
+          throw new RuntimeException(
+              "Failed to create dataset (can not load): " + dbName + "." + tableName);
+        }
+      } catch (TException e) {
+        throw new RuntimeException(e);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    public Dataset register() {
+      try {
+        Table hmsTable = newHmsTable("admin", dbName, tableName, this.location);
+
+        LOGGER.info("Register table: {}.{} in hms.", dbName, tableName);
+        clients.run(
+            client -> {
+              client.createTable(hmsTable);
+              return null;
+            });
+
+        Table hmsTbl = loadHmsTable(dbName, tableName);
+        if (hmsTbl != null) {
+          String dsLocation = Utils.renameScheme(hmsTbl.getSd().getLocation());
+          LOGGER.info("Open table: {} under {}", tableName, location);
+          return Dataset.open(
+              dsLocation, new ReadOptions.Builder().setStorageOptions(this.storageOption).build());
+        } else {
+          throw new RuntimeException(
+              "Failed to create dataset (can not load): " + dbName + "." + tableName);
+        }
+      } catch (TException e) {
+        throw new RuntimeException(e);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    private Table newHmsTable(
+        String hmsTableOwner, String dbName, String tableName, String location)
+        throws TException, InterruptedException {
+      Preconditions.checkNotNull(hmsTableOwner, "'hmsOwner' parameter can't be null");
+      final long currentTimeMillis = System.currentTimeMillis();
+
+      Map<String, String> parameters = Maps.newHashMap();
+      parameters.put(
+          serdeConstants.SERIALIZATION_CLASS,
+          "org.apache.hadoop.hive.serde2.thrift.test.IntString");
+      parameters.put(
+          serdeConstants.SERIALIZATION_FORMAT, "org.apache.thrift.protocol.TBinaryProtocol");
+
+      SerDeInfo serDeInfo =
+          new SerDeInfo(
+              null, "org.apache.hadoop.hive.serde2.thrift.ThriftDeserializer", parameters);
+
+      Database database = clients.run(client -> client.getDatabase(dbName));
+      String tbLocation;
+      if (location == null || location.isEmpty()) {
+        String dbLocation = StringUtils.stripEnd(database.getLocationUri(), "/");
+        tbLocation = String.format("%s/%s%s", dbLocation, tableName, LANCE_DATASET_SUFFIX);
+      } else {
+        tbLocation = location;
+      }
+
+      StorageDescriptor sd =
```

Comment:
do we really need a dummy storage descriptor? I think it can be null in a Hive table